configure_cmd.go

```diff
@@ -0,0 +1,133 @@
+	}
+
+	return checkForUnknownArgs(ctx)
+}
+
+// satelliteAction tries to get bootstrap script from Satellite server and run it.
+// When it is not possible to download the script or running script returns
+// non-zero exit code, then error is returned.
+//
+// It is really risky to download to run some script downloaded from the URL as a
+// root user without any restriction. For this reason, we at least check that
+// provided URL is URL of Satellite server.
+//
+// We would like to use different approach in the future. We would like to use
+// some API endpoints not restricted by username & password for getting CA certs
+// and rendered rhsm.conf, because it would be more secure, but it is not possible ATM
+func satelliteAction(ctx *cli.Context) error {
+	var configureSatelliteResult ConfigureSatelliteResult
+	configureSatelliteResult.format = ctx.String("format")
+	satelliteUrlStr := ctx.String("url")
+
+	uid := os.Getuid()
+	if uid != 0 {
+		errMsg := "non-root user cannot connect system"
+		exitCode := 1
+		return cli.Exit(fmt.Errorf("error: %s", errMsg), exitCode)
+	}
+
+	hostname, err := os.Hostname()
+	if uiSettings.isMachineReadable {
+		configureSatelliteResult.Hostname = hostname
+	}
+	if err != nil {
+		exitCode := 1
+		if uiSettings.isMachineReadable {
+			configureSatelliteResult.HostnameError = err.Error()
+			return cli.Exit(configureSatelliteResult, exitCode)
+		} else {
+			return cli.Exit(err, exitCode)
+		}
+	}
+
+	satelliteUrl, err := normalizeSatelliteScriptUrl(satelliteUrlStr)
+	if err != nil {
+		return fmt.Errorf("could not parse satellite url: %w", err)
+	} else {
+		configureSatelliteResult.SatelliteServerHostname = satelliteUrl.Hostname()
+		configureSatelliteResult.SatelliteServerScriptUrl = satelliteUrl.String()
+	}
+
+	var s *spinner.Spinner = nil
+	if uiSettings.isRich {
+		s = spinner.New(spinner.CharSets[9], 100*time.Millisecond)
+		s.Suffix = fmt.Sprintf(" Configuring '%v' to use Satellite %v", hostname, satelliteUrl.Host)
+		s.Start()
+		// Stop spinner after running function
+		defer func() { s.Stop() }()
+	}
+
+	err = pingSatelliteServer(satelliteUrl, s)
+	if err != nil {
+		return fmt.Errorf("unable to verify that given server is Satellite server: %v", err)
+	} else {
+		configureSatelliteResult.IsServerSatellite = true
+	}
+
+	// Create file for script file
+	satelliteScriptPath := path.Join(varLibDir, "katello-rhsm-consumer")
+	defer os.Remove(satelliteScriptPath)
+	scriptFile, err := os.Create(satelliteScriptPath)
+	if err != nil {
+		return fmt.Errorf("could not create %v file: %w", satelliteScriptPath, err)
+	}
+	defer scriptFile.Close()
+	err = os.Chmod(satelliteScriptPath, 0700)
+	if err != nil {
+		return fmt.Errorf("could not set permissions on %v file: %w", satelliteScriptPath, err)
+	}
+
+	err = downloadSatelliteScript(scriptFile, satelliteUrl, s)
+	if err != nil {
+		return fmt.Errorf("could not download satellite bootstrap script: %w", err)
+	}
+
+	if s != nil {
```

Comment:
Reading over the code, I reached this point and thought to myself "what's `s`?" That means this variable name needs to be more descriptive. It should probably be named `spinner`. Using single letter variable names is convenient, but it's important to consider the lifetime of the variable. If it's going to be long-lived, it's probably better to give it a descriptive name.